pkg/distributor/query.go

```diff
@@ -309,13 +374,29 @@ func (d *Distributor) queryIngesterStream(ctx context.Context, replicationSet ri
 		resp.Timeseries = append(resp.Timeseries, series)
 	}
 
-	reqStats.AddFetchedSeries(uint64(len(resp.Chunkseries) + len(resp.Timeseries)))
-	reqStats.AddFetchedChunkBytes(uint64(resp.ChunksSize()))
-	reqStats.AddFetchedChunks(uint64(resp.ChunksCount()))
+	reqStats.AddFetchedSeries(uint64(len(resp.Chunkseries) + len(resp.Timeseries) + len(resp.StreamingSeries)))
+
+	// Stats for streaming series are handled in streamingChunkSeries.
+	reqStats.AddFetchedChunkBytes(uint64(ingester_client.ChunksSize(resp.Chunkseries)))
+	reqStats.AddFetchedChunks(uint64(ingester_client.ChunksCount(resp.Chunkseries)))
 
 	return resp, nil
 }
 
+// estimatedIngestersPerSeries estimates the number of ingesters that will have chunks for each streaming series.
+func (d *Distributor) estimatedIngestersPerSeries(replicationSet ring.ReplicationSet) int {
+	// Under normal circumstances, a quorum of ingesters will have chunks for each series, so here
+	// we return the number of ingesters required for quorum.
+
+	if replicationSet.MaxUnavailableZones > 0 {
+		// Zone-aware: quorum is number of zones less allowable unavailable zones.
```

Comment:
You may have 4 zones but RF is still 3, so each series will be replicated to 3 (and not 4) ingesters.

```suggestion
		// Zone-aware: quorum is replication factor less allowable unavailable zones.
```